### ISSUE TITLE
Closed curve class

### ISSUE THREAD

**Issue by XorUnison** [ManimCommunity/manim#727]
Manim already has more specific classes for geometry such as `Polygon`. One of my current in-progress PRs adds `ArcPolygon` which is a more general and versatile version of a polygon allowing for arcs. The completely generalized version is a closed curve, for which we should have a `ClosedCurve` eventually, allowing properly filled and animated shapes of arbitrary complexity by taking in functions to draw the boundary.

**Comment by kolibril13** [comment]
This would definitely be a great addition!
E.g. in the use case of intersecting graphs this could be very useful.
At the moment this is done by rectangles, which is not the perfect solution:
https://docs.manim.community/en/latest/examples.html#graphareaplot

**Comment by XorUnison** [comment]
Indeed, that would exactly be the kind of situation `ClosedCurve` could be used for. I take it that `GraphScene` does an internal rectangle schtick? Yeah... that's not ideal at all. I get it for as long as you specifically want to show the rough bars for derivative reasons, but otherwise a proper `ClosedCurve` class could be better used for this and so much more.
